provider/ansible/example.rb

```diff
@@ -33,15 +33,86 @@ module Ansible
       service_account_file: '/tmp/auth.pem'
     }.freeze
 
-    # Holds all information necessary to run gcloud and verify the creation
-    # of a resource.
-    #
-    # command         - The gcloud command being run
-    # failed_name     - The name of the resource being created
-    #                   (defaults to {{resource_name}})
-    #                   Used for verifying deletion.
-    # failed_verifier - A custom Jinja2 verifier to test if gcloud
-    #                   command post-deletion worked as intended.
+    # Examples are used to generate the EXAMPLES block of Ansible documentation
+    # and the integration tests.
+    # Integration tests are a series of YAML tasks (standalone actions).
+    # Integration tests are broken into three parts:
+    # * a list of dependency tasks that should be run.
+    # * a 'task' that is being tested (also used for EXAMPLES block)
+    # * a verifier that will verify cloud status
+    class Example < Api::Object
+      attr_reader :task
+      attr_reader :verifier
+      attr_reader :dependencies
+      attr_reader :facts
+
+      def validate
+        super
+        check_property :task, Task
+        check_optional_property :verifier, Verifier
+        check_optional_property_list :dependencies, Task
+        check_optional_property :facts, Task
+
+        @facts&.set_variable(self, :__example)
+        @verifier.set_variable(self, :__example) if @verifier.respond_to?(:__example)
+      end
+    end
+
+    # A Task represents a single Ansible action. This action is represented
+    # as a standalone YAML block.
+    class Task < Api::Object
+      include Compile::Core
+      attr_reader :name
+      attr_reader :code
+      attr_reader :scopes
+      attr_reader :register
+
+      def validate
+        super
+        check_property :name, String
+        check_property :code, Hash
+        check_optional_property_list :scopes, ::String
+      end
+
+      def build_test(state, object, noop = false)
+        build_task(state, INTEGRATION_TEST_DEFAULTS, object, noop)
+      end
+
+      def build_example(state, object)
+        build_task(state, EXAMPLE_DEFAULTS, object)
+      end
+
+      private
+
+      # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
Add a comment as to why this is disabled?

Reply:
Yep, done!

TL;DR: The method arguments are used in the ERB files. We should figure out a better way to fix this. I know @ndmckinley has talked about it in the past.